rskj-core/src/main/java/co/rsk/peg/bitcoin/BitcoinUtils.java

```diff
@@ -61,4 +62,35 @@ public static Optional<Script> extractRedeemScriptFromInput(TransactionInput txI
             return Optional.empty();
         }
     }
+
+    public static void removeSignaturesFromTransactionWithInputsWithP2shMultiSigInputScript(BtcTransaction transaction) {
+        if (transaction.hasWitness()) {
+            String message = "Removing signatures from SegWit transactions is not allowed.";
+            logger.error(message);
+            throw new IllegalArgumentException(message);
+        }
+
+        if (transaction.getInputs().isEmpty()) {
+            String message = "Cannot remove signatures from transaction inputs of a transaction without inputs.";
+            logger.error(message);
+            throw new IllegalArgumentException(message);
+        }
+
+        for (TransactionInput input : transaction.getInputs()) {
+            Optional<Script> inputRedeemScriptOpt = extractRedeemScriptFromInput(input);
+
+            Script inputRedeemScript = Optional.of(inputRedeemScriptOpt)
+                .get()
+                .orElseThrow(
+                    () -> {
+                        String message = "Cannot remove signatures from transaction inputs that do not have p2sh multisig input script.";
+                        logger.error(message);
+                        return new IllegalArgumentException(message);
+                    }
+                );
```

Comment:
```suggestion
            Script inputRedeemScriptOpt = extractRedeemScriptFromInput(input)
                .orElseThrow(
                    () -> {
                        String message = "Cannot remove signatures from transaction inputs that do not have p2sh multisig input script.";
                        logger.error(message);
                        return new IllegalArgumentException(message);
                    }
                );
```
Can do it directly like that

Reply:
nice, thanks